src/components/Home/PresentRanking/PresentRankingItem.tsx

```diff
@@ -1,80 +1,109 @@
-  const repeatItems = Array.from({ length: repeatCnt }, (_, i) => GOODS_DATA[i % GOODS_DATA.length]);
   const navigate = useNavigate();
 
-  const handleItemClick = (item: Goods) => {
+  const handleItemClick = (item: ProductRanking) => {
     if (!sessionStorage.getItem('email')) {
-      sessionStorage.setItem('redirectProductId', item.id);
+      sessionStorage.setItem('redirectProductId', String(item.id));
       navigate(URLS.login);
     } else {
       navigate(`${URLS.order}?productId=${item.id}`);
     }
   };
 
-  return (
-    <>
-      {repeatItems.map((item: Goods, index: number) => (
-        <div key={index} onClick={() => handleItemClick(item)} style={{ cursor: 'pointer' }}>
-          <StyledPresentRankingItemDiv>
-            <StyledPresentRankingNumContainer index={index + 1}>{index + 1}</StyledPresentRankingNumContainer>
-            <StyledPresentRankingItemImage src={item.imageURL} alt='제품 이미지' />
-            <StyledPresentRankingItemBrandName className='brand_name'>{item.brandInfo.name}</StyledPresentRankingItemBrandName>
-            <StyledPresentRankingItemPresentItem className='goods_name'>{item.name}</StyledPresentRankingItemPresentItem>
-            <StyledPresentRankingItemPrasentPrice className='goods_price'>
-              {item.price.sellingPrice.toLocaleString()} 원
-            </StyledPresentRankingItemPrasentPrice>
-          </StyledPresentRankingItemDiv>
-        </div>
-      ))}
-    </>
-  );
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [isError, setError] = useState<boolean>(false);
+  const [productRanking, setProductRanking] = useState<ProductRankings>({
+    data: [
+      {
+        id: 0,
+        name: '',
+        price: {
+          basicPrice: 0,
+          sellingPrice: 0,
+          discountRate: 0,
+        },
+        imageURL: 'none',
+        brandInfo: {
+          id: 0,
+          name: '',
+          imageURL: 'none',
+        },
+      },
+    ],
+  });
+
+  useEffect(() => {
+    const params = new URLSearchParams(search);
+    const rankType = params.get('rankType');
+    const targetType = params.get('targetType');
+    const typeUrls = `?targetType=${targetType}&rankType=${rankType}`;
+
+    const fetchProductRanking = async () => {
+      try {
+        // const response = await axios.get(process.env.VITE_API_BASE_URL + '/ranking');
+        //const response = await axios.get('http://localhost:3000/api/products/ranking' + typeUrls);
+        const data = await ApiClient('GET', 'products/ranking', null, typeUrls);
+        setProductRanking(data);
+        setError(false);
+      } catch (error) {
+        console.error('Error fetching Product Ranking data:', error);
+        setError(true);
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchProductRanking();
+  }, [isLoading, isError, search]);
+  if (isLoading) {
+    return <div>Loading</div>;
+  } else if (isError || productRanking.data.length <= 0) {
```

Comment:
length는 0보다 작을 수 없으니 아래와 같이 수정하면 좋아보여요

```suggestion
  } else if (isError || productRanking.data.length === 0) {
```